diff --git a/plugins/obs-ndi/CMakeLists.txt b/plugins/obs-ndi/CMakeLists.txt
index 29026e1..4b846d0 100644
--- a/plugins/obs-ndi/CMakeLists.txt
+++ b/plugins/obs-ndi/CMakeLists.txt
@@ -12,6 +12,10 @@ set(LINUX_MAINTAINER_EMAIL "tt2468@irltoolkit.com")
 
 set(OBS_NDI_VERSION "${CMAKE_PROJECT_VERSION}")
 
+if(UNIX AND NOT APPLE)
+    set(OS_LINUX TRUE)
+endif()
+
 # ----------------------------------------------------------------------
 
 target_sources(
@@ -34,20 +38,42 @@ target_sources(
 target_sources(obs-ndi PRIVATE src/forms/SettingsDialog.cpp src/forms/SettingsDialog.h
                                src/forms/resources.qrc)
 
-target_include_directories(obs-ndi PRIVATE ${CMAKE_SOURCE_DIR}/src ${CMAKE_SOURCE_DIR}/lib/ndi)
+if(OS_LINUX)
+  target_include_directories(obs-ndi PRIVATE ${CMAKE_SOURCE_DIR}/src ${CMAKE_SOURCE_DIR}/lib/ndi)
+else()
+  target_include_directories(obs-ndi PRIVATE ${CMAKE_SOURCE_DIR}/src
+                                            ${CMAKE_SOURCE_DIR}/plugins/obs-ndi/lib/ndi
+                                            ${CMAKE_SOURCE_DIR}/UI/obs-frontend-api)
+endif()
 
 # ----------------------------------------------------------------------
 
-find_package(libobs REQUIRED)
+#list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/external")
+#list(APPEND CMAKE_MODULE_PATH "${CMAKE_BINARY_DIR}/UI/obs-frontend-api")
+if(OS_LINUX)
+  list(APPEND CMAKE_PREFIX_PATH "${CMAKE_SOURCE_DIR}/../../build/libobs")
+  find_package(libobs REQUIRED)
+endif()
+
+if(NOT OS_LINUX)
+  list(APPEND CMAKE_PREFIX_PATH "${CMAKE_SOURCE_DIR}/../../build/UI/obs-frontend-api")
+  set(obs-frontend-api_DIR "${CMAKE_BINARY_DIR}/UI/obs-frontend-api")
+endif()
 find_package(obs-frontend-api REQUIRED)
-include(cmake/ObsPluginHelpers.cmake)
+
+if(OS_LINUX)
+  include(cmake/ObsPluginHelpers.cmake)
+endif()
+
 find_qt(COMPONENTS Widgets Core)
 
 configure_file(src/plugin-macros.h.in ${CMAKE_SOURCE_DIR}/src/plugin-macros.generated.h)
 
-target_sources(${CMAKE_PROJECT_NAME} PRIVATE src/plugin-macros.generated.h)
+target_sources(${PROJECT_NAME} PRIVATE src/plugin-macros.generated.h)
 
-target_include_directories(obs-ndi PRIVATE ${CMAKE_SOURCE_DIR}/src)
+if(OS_LINUX)
+  target_include_directories(obs-ndi PRIVATE ${CMAKE_SOURCE_DIR}/src)
+endif()
 
 target_link_libraries(obs-ndi OBS::libobs OBS::obs-frontend-api Qt::Core Qt::Widgets)
 
@@ -63,11 +89,11 @@ target_compile_features(obs-ndi PRIVATE cxx_std_17)
 
 # --- Windows-specific build settings and tasks ---
 if(OS_WINDOWS)
-  configure_file(cmake/bundle/windows/installer-Windows.iss.in
-                 ${CMAKE_BINARY_DIR}/installer-Windows.generated.iss)
+  # configure_file(cmake/bundle/windows/installer-Windows.iss.in
+  #                ${CMAKE_BINARY_DIR}/installer-Windows.generated.iss)
 
   if(MSVC)
-    target_compile_options(${CMAKE_PROJECT_NAME} PRIVATE /W4)
+    target_compile_options(${PROJECT_NAME} PRIVATE /W4)
   endif()
   # --- End of section ---
 
@@ -80,17 +106,24 @@ elseif(OS_MACOS)
   set(MACOSX_PLUGIN_BUNDLE_VERSION "${CMAKE_PROJECT_VERSION}")
   set(MACOSX_PLUGIN_SHORT_VERSION_STRING "1")
 
-  target_compile_options(${CMAKE_PROJECT_NAME} PRIVATE -Wall)
+  target_compile_options(${PROJECT_NAME} PRIVATE -Wall)
 
-  set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES PREFIX "")
+  set_target_properties(${PROJECT_NAME} PROPERTIES PREFIX "")
   # --- End of section ---
 
   # --- Linux-specific build settings and tasks ---
 else()
-  target_compile_options(${CMAKE_PROJECT_NAME} PRIVATE -Wall)
+  target_compile_options(${PROJECT_NAME} PRIVATE -Wall)
 
-  set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES PREFIX "")
+  set_target_properties(${PROJECT_NAME} PROPERTIES PREFIX "")
 endif()
 # --- End of section ---
 
-setup_plugin_target(${CMAKE_PROJECT_NAME})
+if(NOT OS_LINUX)
+  string(REPLACE "-" "_" project_name_without_dash "${PROJECT_NAME}")
+  cpack_add_component_group(${project_name_without_dash}_group
+                            DISPLAY_NAME "obs-ndi plugin"
+                            DESCRIPTION "Network A/V in OBS Studio with NewTek's NDI technology")
+endif()
+
+setup_plugin_target(${PROJECT_NAME})
diff --git a/plugins/obs-ndi/cmake/ObsPluginHelpers.cmake b/plugins/obs-ndi/cmake/ObsPluginHelpers.cmake
index 450c46d..2aa8a55 100644
--- a/plugins/obs-ndi/cmake/ObsPluginHelpers.cmake
+++ b/plugins/obs-ndi/cmake/ObsPluginHelpers.cmake
@@ -491,7 +491,8 @@ else()
       set(OBS_DATA_DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/obs)
     else()
       set(OBS_LIBRARY_DESTINATION bin/${_ARCH_SUFFIX}bit)
-      set(OBS_PLUGIN_DESTINATION obs-plugins/${_ARCH_SUFFIX}bit)
+#      set(OBS_PLUGIN_DESTINATION obs-plugins/${_ARCH_SUFFIX}bit)
+      set(OBS_PLUGIN_DESTINATION obs-plugins)
       set(CMAKE_INSTALL_RPATH "$ORIGIN/" "${CMAKE_INSTALL_PREFIX}/${OBS_LIBRARY_DESTINATION}")
       set(OBS_DATA_DESTINATION "data")
     endif()
@@ -505,7 +506,7 @@ else()
 
       set(CPACK_GENERATOR "DEB")
       set(CPACK_DEBIAN_PACKAGE_DEPENDS
-          "obs-studio (>= 27.0.0), libqt5core5a (>= 5.9.0~beta), libqt5gui5 (>= 5.3.0), libqt5widgets5 (>= 5.7.0)"
+          "obs (>= 1.5.0), libqt5core5a (>= 5.9.0~beta), libqt5gui5 (>= 5.3.0), libqt5widgets5 (>= 5.7.0)"
       )
 
       set(CPACK_OUTPUT_FILE_PREFIX ${CMAKE_SOURCE_DIR}/release)
diff --git a/plugins/obs-ndi/cmake/bundle/macos/installer-macos.pkgproj.in b/plugins/obs-ndi/cmake/bundle/macos/installer-macos.pkgproj.in
index 2a1f643..300f10a 100644
--- a/plugins/obs-ndi/cmake/bundle/macos/installer-macos.pkgproj.in
+++ b/plugins/obs-ndi/cmake/bundle/macos/installer-macos.pkgproj.in
@@ -91,7 +91,7 @@
 											<key>GID</key>
 											<integer>80</integer>
 											<key>PATH</key>
-											<string>obs-studio</string>
+											<string>obs-webrtc</string>
 											<key>PATH_TYPE</key>
 											<integer>2</integer>
 											<key>PERMISSIONS</key>
