diff --git a/plugins/obs-outputs/ftl-sdk/libftl/gettimeofday/gettimeofday.c b/plugins/obs-outputs/ftl-sdk/libftl/gettimeofday/gettimeofday.c
index 08f03fd..3140571 100644
--- a/plugins/obs-outputs/ftl-sdk/libftl/gettimeofday/gettimeofday.c
+++ b/plugins/obs-outputs/ftl-sdk/libftl/gettimeofday/gettimeofday.c
@@ -45,6 +45,7 @@
 
 #ifdef _WIN32
 #include <Windows.h>
+#include <winsock.h>
 
 /* FILETIME of Jan 1 1970 00:00:00. */
 static const unsigned __int64 epoch = ((unsigned __int64)116444736000000000ULL);
