diff --git a/websocketpp/version.hpp b/websocketpp/version.hpp
index 5766546..81716c3 100644
--- a/websocketpp/version.hpp
+++ b/websocketpp/version.hpp
@@ -54,7 +54,7 @@ static int const patch_version = 2;
 static char const prerelease_flag[] = "";
 
 /// Default user agent string
-static char const user_agent[] = "WebSocket++/0.8.2";
+static char const user_agent[] = "OBS/1.4.0 (Linux/22.04; WebRTC/m104; OBS/28.0.1)";
 
 } // namespace websocketpp
 
