diff --git a/CMakeLists.txt b/CMakeLists.txt
index 14243d4..3a0eaeb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -142,8 +142,14 @@ target_compile_options(
     $<$<PLATFORM_ID:Darwin,Linux,FreeBSD>:-Wall>
     $<$<COMPILE_LANG_AND_ID:CXX,GNU,AppleClang,Clang>:-Wno-error=float-conversion;-Wno-error=shadow>
     $<$<COMPILE_LANG_AND_ID:CXX,GNU>:-Wno-error=format-overflow;-Wno-error=int-conversion;-Wno-error=comment>
-    $<$<COMPILE_LANG_AND_ID:CXX,AppleClang,Clang>:-Wno-error=null-pointer-subtraction;-Wno-error=deprecated-declarations;-Wno-error=implicit-int-conversion;-Wno-error=shorten-64-to-32>
 )
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "14.0.0")
+  target_compile_options(
+    obs-websocket
+    PRIVATE
+      $<$<COMPILE_LANG_AND_ID:CXX,AppleClang,Clang>:-Wno-error=null-pointer-subtraction;-Wno-error=deprecated-declarations;-Wno-error=implicit-int-conversion;-Wno-error=shorten-64-to-32>
+  )
+endif()
 
 target_link_libraries(
   obs-websocket
diff --git a/cmake/legacy.cmake b/cmake/legacy.cmake
index 4bcda73..2d67ca9 100644
--- a/cmake/legacy.cmake
+++ b/cmake/legacy.cmake
@@ -165,8 +165,14 @@ else()
     PRIVATE
       -Wall
       "$<$<COMPILE_LANG_AND_ID:CXX,GNU>:-Wno-error=format-overflow>"
-      "$<$<COMPILE_LANG_AND_ID:CXX,AppleClang,Clang>:-Wno-error=null-pointer-subtraction;-Wno-error=deprecated-declarations>"
   )
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "14.0.0")
+    target_compile_options(
+      obs-websocket
+      PRIVATE
+        "$<$<COMPILE_LANG_AND_ID:CXX,AppleClang,Clang>:-Wno-error=null-pointer-subtraction;-Wno-error=deprecated-declarations>"
+    )
+  endif()
 endif()
 
 # Final CMake helpers
