diff --git a/plugins/obs-outputs/websocket-client/websocketpp/websocketpp/version.hpp b/plugins/obs-outputs/websocket-client/websocketpp/websocketpp/version.hpp
index 5766546..4cbb35c 100644
--- a/plugins/obs-outputs/websocket-client/websocketpp/websocketpp/version.hpp
+++ b/plugins/obs-outputs/websocket-client/websocketpp/websocketpp/version.hpp
@@ -54,7 +54,7 @@ static int const patch_version = 2;
 static char const prerelease_flag[] = "";

 /// Default user agent string
-static char const user_agent[] = "WebSocket++/0.8.2";
+static char const user_agent[] = "OBS/1.6.0 (macOS/13.2; WebRTC/m112; OBS/29.1.0)";

 } // namespace websocketpp

